### PR TITLE
Convert to new codebase

### PR DESCRIPTION
**Pull request by aldenlamp**
## Overview

Clean up the codebase to represent the new version of the uplift backend. 

## Changes Made

The new version of uplift (which was previously located inside the src folder of the old version) has now been moved to the main backend. Not a lot has changed from the code perspective other than moving things around and updating requirements.

It might seem like a lot of files have been deleted bc they were part of the old Uplift backend. Since this PR aims to clean up the codebase, those that have been removed in favor of the new uplift backend; those that were useful and not implemented in the new backend will be added back incrementally and tested incrementally. 

In short, everything in src has been deleted or replaced with the contents of src/uplift folder. The main app.py has also been replaced by the app.py inside src/uplift. 

## Test Coverage

I was able to test it (launching from docker) locally to a point where I could pull up the graphql interface in my browser.

### CODE REVIEW

File: requirements.txt [hunk 1]
```diff
@@ -1,49 +1,73 @@
...
+platformdirs==3.10.0
 pre-commit==1.18.3
-promise==2.1
+promise==2.3
 prompt-toolkit==3.0.36
 psycopg2==2.9.5
 ptyprocess==0.7.0
 pure-eval==0.2.2
+pyasn1==0.5.0
+pyasn1-modules==0.3.0
```
Comment (Archit404Error) [inline requirements.txt:52]: Are all of these version bumps and new libraries necessary -- was the requirements file out of date before?

**Review by Archit404Error — COMMENTED** [review]
Only one comment here but again I'm not fully familiar with what the codebase looked like prior to this -- can you please update your PR description to be more detailed and describe what _precisely_ was moved vs deleted? It looks like certain files like scraper.py have been deleted entirely whereas others have only been moved

File: requirements.txt (hunk 1 above)
Comment (aldenlamp) [reply requirements.txt]: I've added some more (hopefully clarifying) text to the PR description. Also I made a commit that updates the requirements only with those that are necessary.

File: app.py [hunk 2]
```diff
@@ -1,15 +1,29 @@
-from datetime import time, datetime
-import json
 from flask import Flask
 from flask_graphql import GraphQLView
+from flask_graphql import GraphQLView
```
Comment (isabellahoie) [inline app.py:3]: Here you have 2 lines of the same code

File: requirements.txt
Comment (isabellahoie) [inline requirements.txt:1]: What are all of these additions?

File: .dockerignore [hunk 3]
```diff
@@ -15,4 +15,6 @@ build/
 
 *.DS_Store
 
-.env
\ No newline at end of file
+.env
+
+Archive
```
Comment (isabellahoie) [inline .dockerignore:20]: What is archive?

**Review by isabellahoie — CHANGES_REQUESTED** [review]
Most of it looks good! Just a couple tiny things that I commented to be changed.

File: requirements.txt
Comment (aldenlamp) [reply requirements.txt:1]: Idt you are reviewing from the correct commit number? Archit mentioned the same concern so I made a new commit that reverts most of the changes in the requirements.

File: .dockerignore (hunk 3 above)
Comment (aldenlamp) [reply .dockerignore:20]: It's just a directory that I'm using locally to make development easier for myself. I can remove it from the gitignore and move that dir somewhere else but I didn't think adding one line to the gitignore would make a difference? lmk if I should still remove it?

**Review by joycrwu — APPROVED** [review]
i'm also not the most familiar with what the codebase looked like prior, but generally everything looks ok with me; if bella says its good to go then all gud

File: src/constants.json [hunk 4]
```diff
@@ -0,0 +1,130 @@
+{
+    "asset_base_url": "https://raw.githubusercontent.com/cuappdev/assets/master/uplift/",
```
Comment (Archit404Error) [inline src/constants.json:2]: I think it's better to store this in `constants.py` instead of the JSON, let's try to centralize all constants/global data in a single location.

File: src/database.sqlite3
Comment (Archit404Error) [inline src/database.sqlite3:1]: Make sure to delete this file

**Review by Archit404Error — APPROVED** [review]
Good to go once you address these changes. My philosophy on constants is just to pull them all into `constants.py` and import selectively. This makes the code more readable and makes it easier to change/read all of the constants via centralization

**Review by isabellahoie — APPROVED** [review]
All good!
